pkg/k8s/policy.go

```diff
@@ -55,47 +52,87 @@ func ServerAuthorizationsForResource(ctx context.Context, k8sAPI *KubernetesAPI,
+				continue
 			}
 			servers = serverList.Items
 		}
 
 		for _, server := range servers {
-			if server.Spec.PodSelector == nil {
-				continue
+			if serverIncludesPod(server, pods) {
+				results = append(results, Authorization{
+					Route:               "",
+					Server:              server.GetName(),
+					ServerAuthorization: saz.GetName(),
+					AuthorizationPolicy: "",
+				})
 			}
+		}
+	}
+
+	policies, err := k8sAPI.L5dCrdClient.PolicyV1alpha1().AuthorizationPolicies(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to get AuthorizationPolicy resources: %s\n", err)
+		os.Exit(1)
+	}
 
-			selector, err := metav1.LabelSelectorAsSelector(server.Spec.PodSelector)
+	for _, p := range policies.Items {
+		target := p.Spec.TargetRef
+		if target.Kind == ServerKind && target.Group == PolicyAPIGroup {
+			server, err := k8sAPI.L5dCrdClient.ServerV1beta1().Servers(p.Namespace).Get(ctx, string(target.Name), metav1.GetOptions{})
 			if err != nil {
-				fmt.Fprintf(os.Stderr, "Failed to create selector: %s\n", err)
-				os.Exit(1)
+				fmt.Fprintf(os.Stderr, "AuthorizationPolicy/%s targets Server/%s but we failed get it: %s\n", p.Name, target.Name, err)
+				continue
 			}
-
-			var selectedPods []corev1.Pod
-			for _, pod := range pods {
-				if selector.Matches(labels.Set(pod.Labels)) {
-					selectedPods = append(selectedPods, pod)
-				}
+			if serverIncludesPod(*server, pods) {
+				results = append(results, Authorization{
+					Route:               "",
+					Server:              server.GetName(),
+					ServerAuthorization: "",
+					AuthorizationPolicy: p.GetName(),
+				})
 			}
+		}
 
-			if serverIncludesPod(server, selectedPods, podSet) {
-				results = append(results, ServerAndAuthorization{server.GetName(), saz.GetName()})
+		if target.Kind == HTTPRouteKind && target.Group == PolicyAPIGroup {
+			route, err := k8sAPI.L5dCrdClient.PolicyV1alpha1().HTTPRoutes(p.Namespace).Get(ctx, string(target.Name), metav1.GetOptions{})
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "AuthorizationPolicy/%s targets HTTPRoute/%s but we failed get it: %s\n", p.Name, target.Name, err)
+				continue
+			}
+			for _, parent := range route.Spec.ParentRefs {
+				if parent.Kind != nil && *parent.Kind == ServerKind &&
+					parent.Group != nil && *parent.Group == PolicyAPIGroup {
+					server, err := k8sAPI.L5dCrdClient.ServerV1beta1().Servers(p.Namespace).Get(ctx, string(parent.Name), metav1.GetOptions{})
+					if err != nil {
+						fmt.Fprintf(os.Stderr, "HTTPRoute/%s belongs to Server/%s but we failed get it: %s\n", target.Name, parent.Name, err)
+						continue
+					}
+					if serverIncludesPod(*server, pods) {
+						results = append(results, Authorization{
+							Route:               route.GetName(),
+							Server:              server.GetName(),
+							ServerAuthorization: "",
+							AuthorizationPolicy: p.GetName(),
+						})
+					}
+				}
```

Comment:
This block looks identical to the above block. Can it be factored out into its own func?

Reply:
They are very similar, but yield different Authorization structs and can print different error messages.  We could factor this, but these blocks are so short that we don't end up saving anything, especially considering the number of arguments we would need to pass in.  The `serverIncludesPod` already factors out the majority of the logic.